0011-interledger-payment-request/0011-interledger-payment-request.md

```diff
@@ -1,140 +1,85 @@
 
-When using the recommended [Condition Generation](#condition-generation) method, account addresses MUST include a request-specific portion to ensure that conditions are unique per-request.
+### Using a PSK Implementation
 
-#### amount
+The [Pre-Shared Key](../0016-pre-shared-key/0016-pre-shared-key.md) protocol uses the payment condition as a MAC of the payment details by setting the fulfillment to be an HMAC of the ILP packet with a key derived from a secret shared between the sender and receiver. In IPR there is no shared secret and the condition is only generated by the receiver. Nevertheless, an IPR receiver can use the same methods to generate the condition and regenerate the fulfillment on receipt of the incoming transfer.
 
-The amount requested by the recipient, such as `10.25`, denoted in the assets of the recipient's ledger.
+To use a PSK implementation, the receiver follows these steps:
 
-The amount MUST be rounded based on the precision supported by the recipient's ledger.
+1. The receiver MAY derive a receiver secret and address as detailed in [PSK: Shared Secret Generation](../0016-pre-shared-key/0016-pre-shared-key.md#shared-secret-generation) but the receiver DOES NOT share the secret key.
+2. The receiver creates the payment packet and condition _using the steps normally executed by the sender_: [PSK: Payment Creation](../0016-pre-shared-key/0016-pre-shared-key.md#1-payment-creation).
+3. The receiver communicates the payment request, including the packet and condition, to the sender.
+4. If the receiver used the derived secret from `1.`, they regenerate that secret from the packet as specified in [PSK: Shared Secret Regeneration](../0016-pre-shared-key/0016-pre-shared-key.md#shared-secret-regeneration).
+5. The receiver generates the fulfillment from the secret and packet as specified in [PSK: Payment Fulfillment](../0016-pre-shared-key/0016-pre-shared-key.md#2-payment-fulfillment).
 
-Recipients SHOULD ignore incoming transfers whose amounts are less than the amount specified in the payment request. Recipients MAY ignore incoming transfers whose amounts are greater than requested.
+### Alternative Methods and General Best Practices
 
-#### condition
+Receivers MAY use any method they choose for generating the condition, such as using a random fulfillment for each request. However, this also requires storing every outstanding fulfillment and packet in a database.
 
-The Crypto Condition to be used as the execution condition for the payment. See [Condition Generation](#condition-generation) for recommendations the recipient MAY use for creating the condition.
+A PSK implementation will already provide these features, but receivers that choose a different method for generating the conditions SHOULD consider these recommendations:
```

Comment:
It's not clear what "these features" means here—it almost reads like a PSK implementation includes storing every outstanding fulfillment in a database. Also, avoid future tense ("will provide") unless you're talking about actual future events.

I suggest rephrasing this section as follows:

> Receivers MAY use any method they choose for generating the condition, such as using a random fulfillment for each request. The receiver MUST be able to identify and verify the ILP Packet for every outstanding payment and produce the corresponding fulfillment. A PSK implementation handles these requirements without storing every fulfillment and packet in a database, by using the condition as a Message Authentication Code (MAC) of the payment request.
> 
> Receivers SHOULD follow these guidelines if implementing a custom system:
>
> - Recivers SHOULD include a nonce in the ILP Packet to distinguish it from other packets that would otherwise be identical. Each payment request SHOULD have a unique nonce.
> - Receivers SHOULD include an expiry date in the ILP Packet, so they do not accept incoming packets that are paid much later than their creation date.